public/app/features/dashboard/dashgrid/DashboardLibrary/CommunityDashboardSection.tsx

```diff
@@ -0,0 +1,306 @@
+import { css } from '@emotion/css';
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom-v5-compat';
+import { useAsync, useDebounce } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Trans, t } from '@grafana/i18n';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { Button, useStyles2, Stack, Grid, EmptyState, Alert, Pagination, FilterInput } from '@grafana/ui';
+import { DataSourceInput } from 'app/features/manage-dashboards/state/reducers';
+
+import { DashboardCard } from './DashboardCard';
+import { MappingContext } from './DashboardLibraryModal';
+import { fetchCommunityDashboard, fetchCommunityDashboards } from './api/dashboardLibraryApi';
+import { DashboardLibraryInteractions } from './interactions';
+import { GnetDashboard } from './types';
+import { tryAutoMapDatasources, parseConstantInputs, isDataSourceInput } from './utils/autoMapDatasources';
+import {
+  getThumbnailUrl,
+  getLogoUrl,
+  buildDashboardDetails,
+  navigateToTemplate,
+} from './utils/communityDashboardHelpers';
+
+interface Props {
+  onShowMapping: (context: MappingContext) => void;
+  datasourceType?: string;
+}
+
+// Constants for community dashboard pagination and API params
+const COMMUNITY_PAGE_SIZE = 9;
+const SEARCH_DEBOUNCE_MS = 500;
+const DEFAULT_SORT_ORDER = 'downloads';
+const DEFAULT_SORT_DIRECTION = 'desc';
+const INCLUDE_LOGO = true;
+const INCLUDE_SCREENSHOTS = true;
+
+export const CommunityDashboardSection = ({ onShowMapping, datasourceType }: Props) => {
+  const [searchParams] = useSearchParams();
+  const datasourceUid = searchParams.get('dashboardLibraryDatasourceUid');
+  const [currentPage, setCurrentPage] = useState(1);
```

Comment:
won't we use a query param for the current page, so if the user returns to the modal they remain in the same page?